src/state/url-sync-mixin.ts

```diff
@@ -62,16 +67,26 @@ export const urlSyncMixin = <
             if (DEBUG) {
               console.log("remove state", ev.detail.dialog);
             }
-            this._ignoreNextPopState = true;
-            mainWindow.history.back();
+            if (history.length) {
+              this._ignoreNextPopState = true;
+              historyPromise = new Promise((resolve) => {
+                historyResolve = () => {
+                  resolve();
+                  historyResolve = undefined;
+                  historyPromise = undefined;
+                };
+              });
+              mainWindow.history.back();
```

Comment:
Since it's not always async, better do it when you know the promise and resolve have been set.

```suggestion
                mainWindow.history.back();
              });
```